R/geom-hex.r

```diff
@@ -56,13 +56,31 @@ geom_hex <- function(mapping = NULL, data = NULL,
 GeomHex <- ggproto("GeomHex", Geom,
   draw_group = function(data, panel_params, coord, lineend = "butt",
                         linejoin = "mitre", linemitre = 10) {
+    if (empty(data)) {
+      return(zeroGrob())
+    }
     if (!inherits(coord, "CoordCartesian")) {
       abort("geom_hex() only works with Cartesian coordinates")
     }
+    # Extract binwidth and height from data if possible
+    if (!is.null(data$width)) {
+      data$xend <- data$x + data$width
```

Comment:
Just so I understand: This is basically a trick to get the width and the height transformed, correct? `xend` and `yend` serve no other purpose otherwise. Have you tried what happens in non-linear coordinates? It'll probably mostly work, but there may be some edge cases where weird stuff happens.

Maybe add a few comments in the code explaining what you're doing, so when the inevitable bug happens 10 years down the road we at least know what the intent was. :-)

Reply:
you understand correctly... I'll add some comments for our future selfs